doc/conf.py

```diff
@@ -255,6 +255,36 @@
     major, minor = match.groups()
     binder_branch = '{}.{}.X'.format(major, minor)
 
+
+class SubSectionTitleOrder:
+    """Sort by title of subsection.
+
+    Assumes README.txt exists for all subsections and uses the subsection with
+    dashes, '---', as the adornment.
+    """
+    def __init__(self, src_dir):
+        self.src_dir = src_dir
+        self.regex = re.compile(r"^([\w ]+)\n-", re.MULTILINE)
+
+    def __repr__(self):
+        return '<%s>' % (self.__class__.__name__,)
```

Comment:
Reading the [docs](https://sphinx-gallery.github.io/configuration.html#sub-gallery-order),

is sphinx going to properly re-build if we add a new folder? The name never changes so sphinx will consider nothing changed.

Not sure that's super critical but maybe we should add a comment

Reply:
sphinx properly rebuilds if there is a new folder in the examples folder.

Reply:
I wanted to subclass:

https://github.com/sphinx-gallery/sphinx-gallery/blob/3dd37f47d76a0e58ac7ff8263882fd59275931b8/sphinx_gallery/sorting.py#L57-L64

But since it was private, I just used its `__repr__`, which is used by the default sorter:

https://github.com/sphinx-gallery/sphinx-gallery/blob/3dd37f47d76a0e58ac7ff8263882fd59275931b8/sphinx_gallery/sorting.py#L112-L125